src/Utils.ts

```diff
@@ -0,0 +1,39 @@
+
+/**
+ * Compares two components.
+ * @remarks
+ *
+ * this recursive function is responsible for comparing two component.
+ *
+ * @param x - the first component that we are going to compare.
+ * @param y - the second component that we are going to compare.
+ * @returns true, if both components are identical; false, if the components are not identical.
+ *
+ */
+const backwardsCheck = (x: any, y: any): boolean => {
+  for (const p in y) {
+    if (Object.prototype.hasOwnProperty.call(y, p) && !Object.prototype.hasOwnProperty.call(x, p)) {return false;}
```

Comment:
Please describe in short comment (above `for...`) why you use the `Object.prototype.hasOwnProperty.call` function not `X.hasOwnProperty(...)`, because in first look it's very hard to understand :) If someone will want to contribute should understand reason through the comment.

Reply:
If I use `X.hasOwnProperty(...)` ESLINT will give the following error.
`ESLint: Do not access Object.prototype method 'hasOwnProperty' from target object.(no-prototype-builtins)`